### DISCUSSION TITLE
Status item state inverted

### DISCUSSION THREAD

**Discussion by jfid** [orchetect/MenuBarExtraAccess#10] [General]
In the Known Issues section you mention:

> There are edge cases where the status item gets confused and may invert its state. It may be related to how the underlying MenuBarExtra works. This is being investigated and a workaround may be possible for a future release.

Here's how to reproduce it (for example with MacControlCenterUI Demo):

1. Click menu bar item to open the main window (the menu bar item will also show a nice background rectangle)
2. Click Settings, main window disappears and menu bar background disappears (all this is OK)
3. Click menu bar item again to open the main window (menu bar item background is shown correctly)
4. Click on any part of the Settings window, and we have the problem: the menu bar main window disappears but the menu bar item background indicates it's active. 

The interesting thing after step 4 is that the main window is **hidden** but **not closed**, so the menu item background is still correct. Getting back the main window to unhidden state needs some clicks in the Desktop, and afterwards everything is back in sync again.

Do you think this behaviour is possible to fix?

I love this package, thanks!

**Comment by orchetect** [comment]
It's extremely difficult to fix and it's on my list to do but I'm not confident a proper solution is even possible.
